R/derive_param_diam_to_sev.R

```diff
@@ -0,0 +1,226 @@
+#'
+#' *Note*: This argument value will be used to check whether input data set has
+#' Severity records for specified `filter_faobj` event. If it has, those records
+#' will be removed and new severity records will be derived from diameters.
+#'
+#' @param test_sev `FATEST` value for severity
+#'
+#' *Default: "Severity"*
+#' *Permitted Value*: A Character scalar
+#'
+#' Assign the value for `FATEST` variable to indicate the severity records.
+#' Ignore the argument if you want to set the default value.
+#'
+#' *Assign the Severity limits to create Severity grade(AVALC)*
+#' @param none limit for grade `"NONE"`
+#'
+#' *Default: c(0,2)*
+#' *Permitted value*: A Numeric vector
+#'
+#' The `none` and the following arguments(`mild`, `mode` and `sev` ) will be
+#' used for assigning the diameter limits to derive the `AVALC`(severity grade).
+#'
+#'  *Note: Use the limit reference to pass the values in argument*
+#'  *Since the condition was coded like this,*
+#'  *NONE : none < AVAL <= none*
+#'  *MILD : mild < AVAL <= mild*
+#'  *MODERATE: mod < AVAL <= mod*
+#'  *SEVERE: sev > AVAL*
+#'  *User should pass the values as numeric vectors. Refer the default values.*
+#'
+#'
+#' @param mild limit for grade `"MILD"`
+#'
+#'  *Default: c(2,5)*
+#'  *Permitted value*: A Numeric vector
+#'
+#'
+#' @param mod limit for grade `"MODERATE"`
+#'
+#' *Default: c(5,10)*
+#' *Permitted value*: A Numeric vector
+#'
+#' @param sev limit for grade `"SEVERE"`
+#'
+#' *Default: 10*
+#' *Permitted value*: A Numeric vector
+#'
+#' @details
+#' 1. Pass the Input data set in `dataset` with required variables and `AVAL`
+#' should have the diameter values for the events(`FAOBJ`) specified in
+#' `filter_faobj`.
+#' 2. If the SDTM data set has severity record for redness and swelling, User
+#' can skip this function to keep the SDTM level severity records for further
+#'  derivation.
+#' 3. If User want to derive the severity records from the diameter records even
+#'  that data set has severity records for Redness and swelling. This function
+#'  will remove the existing severity records and it will create the new
+#'  severity records as`AVALC` which is derived from diameter. and `AVAL` will
+#'  be derived as numeric severity grade to get the maximum severity records
+#'  which will be derived by `derive_param_maxsev.R`.
+#' 4. Pass the values in severity grade arguments(`none`, `mild`, `mod`, `sev`)
+#'  as per the study needs.
+#' 5.User can pass the events which has diameter values to derive their severity
+#'  records by passing the appropriate `filter_diam`, `filter_faobj`and severity
+#'  grade limits.
+#'
+#' @return
+#' The Input data with the new severity records for Redness and swelling which
+#' is specified in `filter_faobj` and AVAL, AVALC will be derived and fatestcd,
+#'  fatest will be changed as per the values
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#' library(admiral)
+#' library(admiraldev)
```

Comment:
You can remove as `admiraldev` is a dependency of `admiral` and will automatically load